### DISCUSSION TITLE
hdf5 error

### DISCUSSION THREAD

**Discussion by christina-rice** [UvaCsl/HemoCell#88] [General discussion]
I am receiving a strange hdf5 error when running example codes. This seems to have occurred on other programs as well, but I am not sure how to specifically debug the HemoCell code as the general solution did not resolve the issue. 

I have followed this [thread](https://github.com/h5py/h5py/issues/1101) suggestion to add to my .bashrc file "export HDF5_USE_FILE_LOCKING='FALSE'"

Additionally, I found this [thread ](https://github.com/ornladios/ADIOS2/issues/757)and this [thread](https://forum.hdfgroup.org/t/hdf5-diag-error-detected-in-hdf5/9517) that suggests changing close()/flush()/MPI_finalize() or adding "--hdf5 …/HDF5" into the script, but I am not sure where to start editing HemoCell to make these changes.

_**Has anyone had this problem? How would you go about resolving it?**_ 

A sample of the error message: 
'''
          (readPositionsBloodCells) Particle count in file (PLT): 80.
          (HemoCell) (Output) writing output at timestep 0 (0 s). Approx. performance: 0.00 s / iteration.
          HDF5-DIAG: Error detected in HDF5 (1.10.8) thread 0:
            # 000: H5F.c line 350 in H5Fcreate(): unable to create file
              major: File accessibility
              minor: Unable to open file
            # 001: H5Fint.c line 1670 in H5F_open(): unable to open file: time = Mon Nov 14 20:09:48 2022
          , name = './output_9//hdf5/000000000000/RBC.000000000000.p.226.h5', tent_flags = 13
              major: File accessibility
              minor: Unable to open file
            # 002: H5FD.c line 723 in H5FD_open(): open failed
              major: Virtual File Layer
              minor: Unable to initialize object
            # 003: H5FDsec2.c line 352 in H5FD__sec2_open(): unable to open file: name = './output_9//hdf5/000000000000/RBC.000000000000.p.226.h5', errno = 2, error message = 'No such file or directory', flags = 13, o_flags = 242
              major: File accessibility
              minor: Unable to open file
          HDF5-DIAG: Error detected in HDF5 (1.10.8) thread 0:
            # 000: H5O.c line 101 in H5Oopen(): not a location
              major: Invalid arguments to routine
              minor: Inappropriate type
            # 001: H5Gloc.c line 228 in H5G_loc(): invalid location ID
              major: Invalid arguments to routine
              minor: Bad value
          HDF5-DIAG: Error detected in HDF5 (1.10.8) thread 0:
            # 000: H5O.c line 101 in H5Oopen(): not a location
              major: Invalid arguments to routine
              minor: Inappropriate type
'''

**Comment by christina-rice** [comment]
I rebuilt HemoCell, and the simulation can run on one node with 36 tasks, but not multiple nodes.

**Reply by mojtabaAmiraslanpour** [reply]
Do you get the same error?
